src/sentry/static/sentry/app/stores/globalSelectionStore.jsx

```diff
@@ -29,18 +30,44 @@ const getDefaultSelection = () => {
   };
 };
 
-const isValidSelection = (selection, organization) => {
-  const allowedProjects = new Set(
-    organization.projects.filter(project => project.isMember).map(p => parseInt(p.id, 10))
-  );
-  if (
-    Array.isArray(selection.projects) &&
-    selection.projects.some(project => !allowedProjects.has(project))
-  ) {
-    return false;
-  }
+const getProjectsByIds = async (organization, projectIds) => {
+  const api = new Client();
+  const query = {};
+  query.query = projectIds.map(id => `id:${id}`).join(' ');
+  return await api.requestPromise(`/organizations/${organization.slug}/projects/`, {
+    query,
+  });
+};
 
-  return true;
+const isValidSelection = async (selection, organization) => {
+  if (organization.projects) {
+    const allowedProjects = new Set(
+      organization.projects
+        .filter(project => project.isMember)
```

Comment:
If your organization has allow-open-join on or the current user has an owner/manager role they can pick projects they are not a member of. You might want to use `project.hasAccess` instead.

Reply:
sounds good this was some older code I moved over, but I'll change it to hasAccess